From 709e8ece1ba3b6a06e70df44b69a29464c86840b Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Fri, 17 Jan 2020 15:02:37 +0200
Subject: [PATCH] [SW-8228] goya2: ppboot: add BTL functionality

Do it at early stage like it was in BTL which executes before ppboot
Now there is no BTL so its PCIe relevant functionality passes here

Change-Id: If3214d8e77a827565f6f2087bfab3f4fc7f919ac
---

diff --git a/goya2_arc/common.c b/goya2_arc/common.c
index 8a49c58..c545c26 100644
--- a/goya2_arc/common.c
+++ b/goya2_arc/common.c
@@ -145,7 +145,33 @@
 	return 1;
 }
 
+void init_prstn(void)
+{
+	/* Configure PRSTN to reset entire SoC */
+	regmap_writel(0xFFFFFFFF, mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L);
+	regmap_writel(0x3FFFFF, mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_H);
+
+	/* Clear masking of PRSTN from SW reset */
+	regmap_writel(0x0, mmPSOC_GLOBAL_CONF_PRSTN_MASK);
+
+	/*
+	 * Hold PCIe controller at reset during PRSTn
+	 * Configure Boot Sequencer to wait for PRSTn
+	 * instead of PCLK stable
+	 */
+	regmap_writel(0x1, mmPSOC_GLOBAL_CONF_PHY_STABLE);
+
+	/*
+	 * Errata: Do not wait for ext_ld_done in boot_seq after
+	 * PRSTn is de-asserted (move to PCIe image without waiting
+	 * for ext_ld_done) ext_ld_done will be configured during
+	 * the PCIe image as part of the FW update flow
+	 */
+	regmap_writel(0x0, mmPSOC_GLOBAL_CONF_BOOT_SEQ_EXT_LD);
+}
+
 void hl_plat_init(void)
 {
-	/* TBD: add relevant code we used to do in BTL */
+	/* Functionality used to be done in BTL */
+	init_prstn();
 }
